contrib/net/http/option.go

```diff
@@ -86,6 +88,14 @@ func WithHeaderTags(headers []string) Option {
 	}
 }
 
+// WithStatusCheck sets a span to be an error if the passed function
+// returns true for a given status code.
+func WithStatusCheck(checker func(statusCode int) bool) Option {
+	return func(cfg *config) {
+		cfg.isStatusError = checker
```

Comment:
Also a nit/not critical, but you might consider changing the parameter name from "checker" to "fn" to conform to other contribs, see: https://github.com/DataDog/dd-trace-go/blob/v1.70.1/contrib/gofiber/fiber.v2/option.go#L87